100.md

```diff
@@ -0,0 +1,528 @@
+# NIP-100
+
+## Android Signer Application
+
+`draft` `optional`
+
+This NIP describes a method for 2-way communication between an Android signer and any Nostr client on Android. The Android signer is an Android Application and the client can be a web client or an Android application.
+
+# Usage for Android applications
+
+The Android signer uses Intents and Content Resolvers to communicate between applications.
+
+To be able to use the Android signer in your application you should add this to your AndroidManifest.xml:
+
+```xml
+<queries>
+  <intent>
+    <action android:name="android.intent.action.VIEW" />
+    <category android:name="android.intent.category.BROWSABLE" />
+    <data android:scheme="nostrsigner" />
+  </intent>
+</queries>
+```
+
+Then you can use this function to check if there's a signer application installed:
+
+```kotlin
+fun isExternalSignerInstalled(context: Context): Boolean {
+  val intent =
+    Intent().apply {
+      action = Intent.ACTION_VIEW
+      data = Uri.parse("nostrsigner:")
+    }
+  val infos = context.packageManager.queryIntentActivities(intent, 0)
+  return infos.size > 0
+}
+```
+
+## Using Intents
+
+To get the result back from the Signer Application you should use `registerForActivityResult` or `rememberLauncherForActivityResult` in Kotlin. If you are using another framework check the documentation of your framework or a third party library to get the result.
```

Comment:
I personally would appreciate a simple example for using `rememberLauncherForActivityResult`. It would have made things easier for me :D


```kotlin
val launcher = rememberLauncherForActivityResult(contract = ActivityResultContracts) { activityResult ->
  val signature = activityResult.data?.getStringExtra("signature")
  // Do something with signature ...
}

launcher.launch(input = intent, options = ActivityOptionsCompat.makeBasic())
```